scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -87,9 +87,14 @@ object ScioContext {
     new ScioContext(options, artifacts)
 
   /** Create a new [[ScioContext]] instance for testing. */
-  def forTest(): ScioContext = {
+  def forTest(jobName: String = ""): ScioContext = {
```

Comment:
@ravwojdyla @nevillelyh want to get your feedback especially here. jobName is a dataflow specific option so not sure if it makes sense here. Supporting user-defined job names from the test context isn't really that important and I can get rid of this if you think it's better without.